lib/common/benchmarkModelGenerator.js

```diff
@@ -0,0 +1,408 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const { Blob } = require('buffer');
+const {
+    uniqueNamesGenerator, adjectives, colors, animals
+} = require('unique-names-generator');
+
+const DEFAULT_CONCERTO_METAMODEL_VERSION = '1.0.0';
+const DEFAULT_MODEL_NAMESPACE = 'big.benchmark.model';
+const DEFAULT_MODEL_VERSION = '1.0.0';
+
+/**
+ * Generate a benchmark model.
+ *
+ * @private
+ * @class
+ */
+class BenchmarkModelGenerator {
+    /**
+     * Formats the given byte size into a human-readable string.
+     * @param {number} bytes The number of bytes.
+     * @param {number} [decimals=2] The number of decimal places to include.
+     * @returns {string} A formatted string representing the byte size.
+     */
+    formatBytes(bytes, decimals = 2) {
+        const k = 1024;
+        const dm = decimals < 0 ? 0 : decimals;
+        const sizes = ['Bytes', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
```

Comment:
We would probably run out of disk space if we try to generate a petabyte model :D

Reply:
The sky's the limit! ☁️

Reply:
A practical limit will be the heap space allocated to the process given that we're not doing any streaming here. It should be cheap to check that the requested bytes is smaller than the available heap size? 
https://nodejs.org/api/v8.html#v8getheapstatistics
`total_available_size` appears to be what we need.

Reply:
Addressed ✅